src/coreclr/System.Private.CoreLib/src/System/Threading/Thread.CoreCLR.cs

```diff
@@ -387,6 +415,9 @@ public void DisableComObjectEagerCleanup() { }
         /// </summary>
         public void Interrupt()
         {
+#if TARGET_UNIX || TARGET_BROWSER || TARGET_WASI
+            WaitSubsystem.Interrupt(this);
+#endif
             Interrupt(GetNativeHandle());
```

Comment:
Should this be under `#else`? Or is this still doing something even with managed wait system?